.github/workflows/tests.yaml

```
@@ -0,0 +1,31 @@
+name: run
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: unit-tests
+    runs-on: ubuntu-latest
```

You can also test for macos as:

```suggestion
    strategy:
      matrix:
        platform: [ubuntu-latest, macos-latest]
    runs-on: ${{ matrix.platform }}
```